NSS-Spontaneous-Spoonbills/Sprint2
# Dp department view

## Link to Ticket

#5 

Closes #

doesn't close the ticket, test suit needs to be built for it next

## Description of Proposed Changes
1. adds department view
1. adds templates to display list of departments

## Steps to Test

1. cd to ../sprint2
```
git fetch --all
git checkout dp_department_view
rm -rf db.sqlite3
cd WorkforceManagement/migrations
ls
```
1. nuke everything in migrations besides __init__.py and __pycache__

```
../../
python manage.py makemigrations WorkforceManagement
python manage.py migrate
python manage.py runserver
```
1. open db and add some entries into the departments table 
1. navigate to http://localhost:8000/WorkforceManagement/departments/ and see if the departments are listed. also try clicking on other nav bar elements and making sure that when you click back on departments that the departments that you added appear on the list

## Impacted Areas in Application
in /WorkforceManagement/
* urls.py
* views/Department_View.py
* views/__init__.py
*templates/WorkforceManagement/Department_list.html

## Mentions @username

@binauralbeat @JacobAndrewSmith92 @MrErin 


## Testing




## Deploy Notes

Notes regarding deployment the contained body of work.  These should note any
db migrations, etc.

## Definition of Done

1. The project must be fully documented. This includes the following:
   1. Complete README that documents the steps to install the code, how to install any dependencies, or system configuration needed.
   1. Every class must be documented with purpose, author, and methods.
   1. Every method must be documented with purpose and argument list - which itself must contain a short purpose for each argument.
1. The project must be able to run fully, and without errors, on each teammate's system.
1. Fulfills every requirement.
1. Every line of code has been peer reviewed.
1. For projects that require unit testing, core functionality must be identified and have at least one test for each
